src/main/java/com/livingobjects/neo4j/ExportExtension.java

```diff
@@ -318,9 +321,39 @@ private void initializePropertiesToExport(Lineages lineages, Lineage lineage) {
         }
     }
 
-    private PaginatedLineages extract(FullQuery fullQuery) {
+    private PaginatedLineages extract(FullQuery initQuery) {
         try (Transaction tx = graphDb.beginTx()) {
             MetaSchema metaSchema = new MetaSchema(tx);
+
+            // to be sure we isolate scopes, always add a filter on scope for each individual query
+        List<ExportQuery> queriesWithScopeFilter = initQuery.exportQueries.stream()
+                .map(q -> {
+                    // any requiredAttributes is neithre global or sp
+                    boolean isClientScope = q.requiredAttributes.stream()
+                            .map(attr -> metaSchema.getAuthorizedScopes(tx, attr))
+                            .filter(scope -> !scope.contains(SCOPE_GLOBAL_ATTRIBUTE))
+                            .anyMatch(s -> true);
+                    if (q.scopes.isEmpty() || !isClientScope) {
+                        return q;
+                    }
+
+                    List<Filter<Column>> scopeFilters = q.scopes.stream()
+                            // remove global scope
+                            .filter(s -> !s.equals(GLOBAL_SCOPE.id) && !s.equals(SP_SCOPE.id))
+                            // remove scopes already filtered
+                            .filter(s -> !getScopeFilters(q).anyMatch(f -> f.valueFilter.value.toString().equals(s)))
```

Comment:
extract getScopeFilters(q) outside of stream loop, maybe even using a set for efficient contains